src/figuresClasses.ts

```diff
@@ -2,10 +1,79 @@
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides = [a, b, c];
+    const maxSide = Math.max(a, b, c);
+    const leftSides = sides.filter((side) => side !== maxSide);
+    const sumLeftSides = leftSides.reduce((acc, side) => acc + side);
+
+    if (a <= 0 || b <= 0 || c <= 0 || sumLeftSides <= maxSide) {
+      throw new Error('your error message');
+    }
+  }
+
+  getArea(): number {
+    const semiPerimetr = (this.a + this.b + this.c) / 2;
+
+    return (
+      Math.floor(
+        Math.sqrt(
+          semiPerimetr *
+            (semiPerimetr - this.a) *
+            (semiPerimetr - this.b) *
+            (semiPerimetr - this.c),
+        ) * 100,
+      ) / 100
+    );
+  }
+}
+
+export class Circle implements Figure {
+  shape: string = 'circle';
+
+  constructor(
+    public color: string,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
```

Comment:
The error message should be more descriptive. Consider using a message like 'Radius must be greater than zero'.